steps/s3/session.go

```diff
@@ -0,0 +1,179 @@
+// Copyright 2021 Telefonica Cybersecurity & Cloud Tech SL
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package s3steps
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/Telefonica/golium"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	aws_s "github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+	"github.com/sirupsen/logrus"
+)
+
+type Session struct {
+	Client           *aws_s.Session
+	CreatedDocuments []*CreatedDocument
+}
+
+type CreatedDocument struct {
+	bucket string
+	key    string
+}
+
+// newS3Session initiates a new aws session.
+func (s *Session) newS3Session(ctx context.Context) error {
+	logrus.Debugf("Creating a new S3 session")
+	var err error
+
+	s3Config := &aws.Config{
+		Region: aws.String("eu-west-1"),
+	}
+	// Check minio
+	minio := golium.Value(ctx, "[CONF:minio]").(bool)
+	if minio {
+		s3Config = &aws.Config{
+			Credentials:      credentials.NewStaticCredentials(golium.Value(ctx, "[CONF:minio-aws-access-key-id]").(string), golium.Value(ctx, "[CONF:minio-aws-secret-access-key]").(string), ""),
+			Endpoint:         aws.String(golium.Value(ctx, "[CONF:minio-endpoint]").(string)),
+			Region:           aws.String(golium.Value(ctx, "[CONF:minio-aws-region]").(string)),
+			DisableSSL:       aws.Bool(true),
+			S3ForcePathStyle: aws.Bool(true),
+		}
+	}
```

Comment:
Ain't it be possible to have an step to create a new session with parameters instead similar to steps in elasticsearch or redis?

Reply:
I dont think it'd be usefull due all the parameters will be obtained from env: AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY, AWS_REGION
This "if minio"block is just a trick to run golium ATs

Reply:
But in case you want to override the configuration from the environment variables it may be useful. Think that for minio you should override them because the need other configurations from the present in the AWS environment variables.